cpp/demo/poisson/main.cpp

```diff
@@ -189,6 +190,10 @@ int main(int argc, char* argv[])
       [](auto values, auto x) { values = Eigen::sin(5 * x.col(0)); });
   L->set_coefficients({{"f", f}, {"g", g}});
 
+  // Prepare and set Constants for the bilinear form
+  std::shared_ptr<function::Constant> kappa(new function::Constant({2.0}));
```

Comment:
Can we use `auto kappa = std::make_shared<function::Constant>(2.0);`?

Reply:
Yes, but only by adding another constructor to `Constant`. `kappa` is in general, vector valued, so makes sense to have it as `auto kappa = std::make_shared<function::Constant>({2.0});` but this doesn't work due to "initialiser_list" issues in C++.

Reply:
@chrisrichardson what do you mean by "is in general, vector valued,"?